homeassistant/components/climate/intent.py

```diff
@@ -52,3 +61,84 @@ async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse
         response.response_type = intent.IntentResponseType.QUERY_ANSWER
         response.async_set_states(matched_states=match_result.states)
         return response
+
+
+class SetTemperatureIntent(intent.IntentHandler):
+    """Handle SetTemperature intents."""
+
+    intent_type = INTENT_SET_TEMPERATURE
+    description = "Sets the target temperature of a climate device or entity"
+    slot_schema = {
+        vol.Required("temperature"): vol.Coerce(float),
+        vol.Optional("area"): intent.non_empty_string,
+        vol.Optional("name"): intent.non_empty_string,
+        vol.Optional("floor"): intent.non_empty_string,
+        vol.Optional("preferred_area_id"): cv.string,
+        vol.Optional("preferred_floor_id"): cv.string,
+    }
+    platforms = {DOMAIN}
+
+    async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
+        """Handle the intent."""
+        hass = intent_obj.hass
+        slots = self.async_validate_slots(intent_obj.slots)
+
+        temperature: float = slots["temperature"]["value"]
+
+        name: str | None = None
+        if "name" in slots:
+            name = slots["name"]["value"]
+
+        area_name: str | None = None
+        if "area" in slots:
+            area_name = slots["area"]["value"]
+
+        floor_name: str | None = None
+        if "floor" in slots:
+            floor_name = slots["floor"]["value"]
+
+        match_constraints = intent.MatchTargetsConstraints(
+            name=name,
+            area_name=area_name,
+            floor_name=floor_name,
+            domains=[DOMAIN],
+            assistant=intent_obj.assistant,
+            features=ClimateEntityFeature.TARGET_TEMPERATURE,
+            single_target=True,
+        )
+        match_preferences = intent.MatchTargetsPreferences(
+            area_id=slots.get("preferred_area_id", {}).get("value"),
+            floor_id=slots.get("preferred_floor_id", {}).get("value"),
+        )
+        match_result = intent.async_match_targets(
+            hass, match_constraints, match_preferences
+        )
+        if not match_result.is_match:
+            raise intent.MatchFailedError(
+                result=match_result, constraints=match_constraints
+            )
+
+        assert match_result.states
+        climate_state = match_result.states[0]
```

Comment:
I'm not sure, if we should limit to the first found climate entity - what if I have 2 or 3 climates in the same room (_which is usually represented by an area_)?

Reply:
I think that `single_target` on `MatchTargetsConstraints` would make the match fail if there is more than 1 match.

Reply:
The `single_target` constraint will cause a match failure if more than one climate entity matches. We'll need to add error messages for this eventually.